utils/common/src/main/java/org/apache/brooklyn/util/text/NaturalOrderComparator.java

```diff
@@ -140,6 +147,32 @@ public int compare(String a, String b) {
                 if ((result = compareRight(a.substring(ia), b.substring(ib))) != 0) {
                     return result;
                 }
+                // numeric portion is the same; previously we incremented and checked again
```

Comment:
i noticed some behaviour which i thought was undesired behaviour of `NaturalOrderComparator`, as indicated by the new tests